e2e/action-row.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { test, expect, type Page, type Locator } from '@playwright/test';
+
+const challengeButtons = [
+  'Instructions',
+  'index.html',
+  'styles.css',
+  'Console'
+];
+
+const editorButtons = ['index.html', 'styles.css'];
+
+let page: Page;
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto(
+    '/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form'
+  );
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
```

Comment:
```suggestion
test.beforeEach(async ({ page }) => {
  await page.goto(
    '/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form'
  );
});

test.afterEach(async ({ page }) => {
  await page.close();
});
```

It's ideal to run the tests in isolation.